tests/test_intopydict.rs

```diff
@@ -0,0 +1,74 @@
+#![cfg(feature = "macros")]
+
+use pyo3::{prelude::IntoPyDict, pyclass, types::IntoPyDict};
+
+pub trait TestTrait<'a> {}
+
+#[pyclass]
+#[derive(IntoPyDict)]
+pub struct TestDict {
+    x: u8,
+}
+
+#[derive(IntoPyDict)]
+pub struct PyClass {
+    x: u8,
+    y: TestDict,
+}
+
+#[derive(IntoPyDict, PartialEq, Debug, Clone)]
+pub struct Test1 {
+    x: u8,
+}
+
+#[derive(IntoPyDict, Clone, Debug)]
+pub struct Test {
+    #[pyo3(get, set, name = "hello")]
+    v: Vec<Vec<Test1>>,
+    j: Test1,
```

Comment:
Potentially add a `#[cfg(FALSE)] z: i32` field to check interaction with `#[cfg]`.